crates/matrix-sdk-crypto/src/error.rs

```diff
@@ -70,8 +72,26 @@ pub enum OlmError {
             have a valid Olm session with us"
     )]
     MissingSession,
+
+    #[error(transparent)]
+    /// The room key that was to be shared was not shared because the sharing
+    /// strategy could not be fulfilled.
+    RoomKeySharingStrategyError(DeviceCollectError),
 }
 
+/// Depending on the sharing strategy for room keys, the distribution of the
+/// room key could fail.
+#[derive(Error, Debug)]
+pub enum DeviceCollectError {
+    /// When a verified user has some unsigned devices.
+    /// Happens only with `CollectStrategy::DeviceBasedStrategy` when
+    /// `error_on_unsigned_device_of_verified_users` is true.
+    ///
+    /// In order to resolve this the caller has to ignore or blacklist the given
+    /// devices (see [`Device::set_local_trust`]).
+    #[error("Encryption failed because one or several verified users have unsigned devices")]
```

Comment:
For the record: error messages start with a lower-case letter, so that they can be more easily composed.